src/components/ec-currency-input/ec-currency-input.spec.js

```diff
@@ -0,0 +1,160 @@
+
+      wrapper.setData({ locale: 'en' });
+      expect(wrapper.findByDataTest('ec-currency-input__amount').element.value).toEqual('1,111.11');
+    });
+  });
+
+  describe('@events', () => {
+    it('should emit change event when an item is selected', () => {
+      const wrapper = mountCurrencyInput();
+
+      selectItem(wrapper, 1);
+      expect(wrapper.emitted('change').length).toEqual(1);
+      selectItem(wrapper, 2);
+      expect(wrapper.emitted('change').length).toEqual(2);
+    });
+
+    it('should emit change event when amount is set', async () => {
+      const wrapper = mountCurrencyInput();
+
+      wrapper.findByDataTest('ec-currency-input__amount').setValue('11');
+      await wrapper.vm.$nextTick();
+      expect(wrapper.emitted('change').length).toEqual(1);
+      wrapper.findByDataTest('ec-currency-input__amount').setValue('111');
+      await wrapper.vm.$nextTick();
+      expect(wrapper.emitted('change').length).toEqual(2);
+    });
+  });
+
+  describe('v-model', () => {
+    it('should use the v-model with the currency and emit the changes', () => {
+      const wrapper = mountCurrencyInputAsTemplate(
+        '<ec-currency-input :currencies="currencies" v-model="value" />',
+        {},
+        {
+          data() {
+            return { currencies, value: { currency: '', amount: 0 } };
+          },
+        },
+      );
+
+      selectItem(wrapper, 0);
+      expect(wrapper.vm.value.currency).toEqual(currencies[0]);
+      selectItem(wrapper, 1);
+      expect(wrapper.vm.value.currency).toEqual(currencies[1]);
+    });
+
+    it('should use the v-model with the amount and emit the changes', async () => {
+      const wrapper = mountCurrencyInputAsTemplate(
+        '<ec-currency-input :currencies="currencies" v-model="value" />',
+        {},
+        {
+          data() {
+            return { currencies, value: { amount: 0 } };
+          },
+        },
+      );
+
+      wrapper.findByDataTest('ec-currency-input__amount').setValue('11');
+      await wrapper.vm.$nextTick();
+      expect(wrapper.vm.value.amount).toEqual(11);
+    });
+
+    it('should not show decimal when currency is JPY', async () => {
+      const wrapper = mountCurrencyInputAsTemplate(
+        '<ec-currency-input :currencies="currencies" v-model="value" />',
+        {},
+        {
+          data() {
+            return { currencies, value: { currency: 'GBP', amount: 11111.11 } };
+          },
+        },
+      );
+
+      expect(wrapper.findByDataTest('ec-currency-input__amount').element.value).toEqual('11,111.11');
+      selectItem(wrapper, 3);
+      await wrapper.vm.$nextTick();
+      expect(wrapper.findByDataTest('ec-currency-input__amount').element.value).toEqual('11,111');
+    });
+
+    // test JPY currency
+  });
+});
+
+function selectItem(wrapper, index) {
+  wrapper.findAll('.ec-dropdown-search__item').at(index).trigger('click');
```

Comment:
Each item in the dropdown has data-test like this:
```
:data-test="`ec-dropdown-search__item ec-dropdown-search__item--${index}`"
```
so you can change this to:
```js
wrapper.findByDataTest(`ec-dropdown-search__item--${index}`).trigger('click');
```